server/src/main/java/org/elasticsearch/index/engine/ReadOnlyEngine.java

```diff
@@ -220,7 +220,8 @@ protected DirectoryReader open(IndexCommit commit) throws IOException {
         assert Transports.assertNotTransportThread("opening index commit of a read-only engine");
         DirectoryReader directoryReader = DirectoryReader.open(
             commit,
-            org.apache.lucene.util.Version.MIN_SUPPORTED_MAJOR,
+            // Temporary workaround
+            config().getIndexSettings().getIndexMetadata().getCompatibilityVersion().luceneVersion().major,
```

Comment:
I don't know if I would use the compatibility version. The end result is the same, but perhaps I would do what you did above, based on MINIMUM_READONLY_COMPATIBLE.luceneVersion().major . There is no need to make this smarter, it is just a lower bound, possibly safety around making sure we never open indices that are older than the minimum version supported.

Reply:
That makes sense, I changed this to `MINIMUM_READONLY_COMPATIBLE.luceneVersion().major` as you suggested, thanks.